cmake/CMakeLists.txt

```diff
@@ -411,9 +415,21 @@ if (NOT (UNIX AND onnxruntime_ENABLE_PYTHON AND onnxruntime_ENABLE_TRAINING AND
   set(onnxruntime_ENABLE_TRAINING_TORCH_INTEROP OFF)
 endif()
 
+if (NOT (UNIX AND onnxruntime_USE_CUDA AND onnxruntime_ENABLE_PYTHON AND onnxruntime_ENABLE_TRAINING AND (NOT onnxruntime_BUILD_SHARED_LIB)))
+  if (onnxruntime_ENABLE_TRITON)
+    message(WARNING "onnxruntime_ENABLE_TRITON is turned OFF due to incompatible build combinations.")
+  endif()
+  set(onnxruntime_ENABLE_TRITON OFF)
+endif()
+
 set(onnxruntime_REQUIRE_PYTHON_EMBED_LIB OFF)
-if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
-  add_compile_definitions(ENABLE_TRAINING_TORCH_INTEROP)
+if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP OR onnxruntime_ENABLE_TRITON)
+  if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
+    add_compile_definitions(ENABLE_TRAINING_TORCH_INTEROP)
+  endif()
+  if (onnxruntime_ENABLE_TRITON)
```

Comment:
nit: this can be a separate if, does not need to be nested under the if on line 426